.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,15 @@
+<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change? -->
+## Summary of the Pull Request
+
+<!-- Other than the issue solved, is this relevant to any other issues/existing PRs? --> 
+## References
+
+<!-- Please review the items on the PR checklist before submitting-->
+## PR Checklist
+[ ] closes #xxx
+[ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
+[ ] tests added/passed
+[ ] requires documentation to be updated
```

Comment:
@DHowett-MSFT do we want to add a checkbox about "I've discussed this with the core contributors prior to initiating the request. If this box is not checked, I'm ready to accept this work might be discarded in favor of a different grand plan."

I know we haven't done #750 yet, but we wanted to create some sort of guidance that will prevent unexpected hurt if we have to close something because it's not in our master plan and we didn't talk about it with the person yet (or manage to get around to fully documenting our plan on GitHub yet.)

Reply:
I support that. We can also evolve this guidance over time as we work on #750.

You should use the "Suggestion" feature, and @Bharat123rox can apply it as a change to their pull request from the github UI directly!

![image](https://user-images.githubusercontent.com/14316954/57711618-e6d28500-7623-11e9-95f8-2182da2eb502.png)

Reply:
Good suggestion. I need to learn2github

Reply:
```suggestion
[ ] Requires documentation to be updated
[ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan.
```

Reply:
```suggestion
[ ] Requires documentation to be updated
[ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #___
```